### DISCUSSION TITLE
Multilingual sites

### DISCUSSION THREAD

**Discussion by bkmgit** [mmistakes/minimal-mistakes#3499] [Ideas]
### Motivation

Documentation on internationalization is limited.  Adding examples of multilingual sites would be helpful.

### Suggestion

- Add a link to https://github.com/Leo3418/leo3418.github.io and possibly a description of one way to make a multilingual site similar to https://leo3418.github.io/collections/multilingual-jekyll-site which uses [polyglot](https://github.com/untra/polyglot)

- Possibly discuss other options such as [Jekyll-multiple-languages plugin](https://github.com/kurtsson/jekyll-multiple-languages-plugin) and/or [Jekyll-I18n](https://github.com/ruby-gettext/jekyll-task-i18n), brief overview given at https://gist.github.com/pavanjoshi914/1cebce0f55539be221473abd04aaceaf

**Comment by PanderMusubi** [comment]
I am trying a minimal example according to your example. With jekyll-polyglot (1.9.0) and minimal-mistakes-jekyll (4.27.3) I get the following errors:

    Run in verbose mode to see all warnings.
    test/vendor/bundle/ruby/3.0.0/gems/jekyll-polyglot-1.9.0/lib/jekyll/polyglot/patches/jekyll/site.rb:245:in `relative_url_regex': target of repeat operator is not specified: /href="?\\/((?:(?!*.sh)(?!.sass-cache)(?!.jekyll-cache)(?!gemfiles)(?!Gemfile)(?!Gemfile.lock)(?!node_modules)(?!vendor\\/bundle\\/)(?!vendor\\/cache\\/)(?!vendor\\/gems\\/)(?!vendor\\/ruby\\/)(?!javascript)(?!images)(?!css)(?!public)(?!sitemap)(?!en\\/)(?!es\\/)(?!nl\\/)[^,'"\\s\\/?.]+\\.?)*(?:\\/[^\\]\\[)("'\\s]*)?)"/ (RegexpError)
    	from test/vendor/bundle/ruby/3.0.0/gems/jekyll-polyglot-1.9.0/lib/jekyll/polyglot/patches/jekyll/site.rb:204:in `process_documents'
    	from test/vendor/bundle/ruby/3.0.0/gems/jekyll-polyglot-1.9.0/lib/jekyll/polyglot/hooks/process.rb:8:in `block in hook_process'
    	from test/vendor/bundle/ruby/3.0.0/gems/jekyll-polyglot-1.9.0/lib/jekyll/polyglot/hooks/process.rb:7:in `each'
    	from test/vendor/bundle/ruby/3.0.0/gems/jekyll-polyglot-1.9.0/lib/jekyll/polyglot/hooks/process.rb:7:in `hook_process'
    	from test/vendor/bundle/ruby/3.0.0/gems/jekyll-polyglot-1.9.0/lib/jekyll/polyglot/hooks/process.rb:3:in `block in <top (required)>'
    	from test/vendor/bundle/ruby/3.0.0/gems/jekyll-4.4.1/lib/jekyll/hooks.rb:103:in `block in trigger'
    ...

How to fix this?

**Reply by PanderMusubi** [reply]
Solved it, my OS needs higher version of Ruby, then the minimally 1.11.0 of jekyll-polyglot where it is fixed will be installed.

**Comment by PanderMusubi** [comment]
I have made support for Polyglot, the test site is here. https://github.com/PanderMusubi/jekyll-minimal-mistakes-polyglot

Please review it so it can be converted into a PR. The changes are minimal, but the support for multilingual sites jumps forwards.

**Reply by mattions** [reply]
I've used your improvements on my site and it worked very well.

What I do not understand though, is how to make the pagination work properly for both languages.

When I launch jekyll I get index is missing

The repo is at:, just in case you want to take a look: https://github.com/ippocra/ippocra.github.io

I would like to get the paginate-v2 working

Nice touch the language selector in the footer, BTW!
